js/client/modules/@arangodb/aql/functions.js

```diff
@@ -0,0 +1,131 @@
+/* jshint strict: false */
+/* global db */
+
+// //////////////////////////////////////////////////////////////////////////////
+// / @brief AQL user functions management
+// /
+// / @file
+// /
+// / DISCLAIMER
+// /
+// / Copyright 2012 triagens GmbH, Cologne, Germany
+// /
+// / Licensed under the Apache License, Version 2.0 (the "License")
+// / you may not use this file except in compliance with the License.
+// / You may obtain a copy of the License at
+// /
+// /     http://www.apache.org/licenses/LICENSE-2.0
+// /
+// / Unless required by applicable law or agreed to in writing, software
+// / distributed under the License is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / Copyright holder is triAGENS GmbH, Cologne, Germany
+// /
+// / @author Jan Steemann
+// / @author Copyright 2012, triAGENS GmbH, Cologne, Germany
+// //////////////////////////////////////////////////////////////////////////////
+
+var internal = require('internal');
+var arangosh = require('@arangodb/arangosh');
+
+var ArangoError = require('@arangodb').ArangoError;
+
+// //////////////////////////////////////////////////////////////////////////////
+// / @brief validate a function name
+// //////////////////////////////////////////////////////////////////////////////
+
+var validateName = function (name) {
```

Comment:
please remove. client should not do any validation. that way we have the validation just once on the server-side and do not need to maintain it in two places.